test/pynrn/test_swc.py

```diff
@@ -0,0 +1,135 @@
+from neuron import h, gui
+h.load_file("import3d.hoc")
+
+tst_data = ['''
+# One-point soma
+#n,type,x,y,z,radius,parent
+1  1 0 0 0 10 -1
+2  3 0 10 0 1 1
+3  3 0 20 0 1 2
+4  3 0 -10 0 2 1
+5  3 0 -20 0 2 4
+6  3 10 0 0 .5 1
+7  3 10 5 0 1 6
+''','''
+# Multi-point soma
+#n,type,x,y,z,radius,parent
+1  1 0 0 0 10 -1
+2  1 2 0 0 10  1
+3  1 4 0 0 10  2
+4  3 2 10 0   1  2
+5  3 2 18 0  1  4
+''','''
+# dendrites at ends of multisection soma
+#n,type,x,y,z,radius,parent
+1  1 0 0 0 5 -1
+2  1 10 0 0 3 1
+3  1 -8 -6 0 3 1
+4  1 -8 6 0 3 1
+5  3 20 0 0 1 2
+6  3 -16 -12 0 1 3
+7  3 -16 12 0 1 4
+''','''
+#two section soma with dendrites at ends and off point 1
+#n,type,x,y,z,radius,parent
+1 1 0 0 0 5 -1
+2 1 5 0 0 3 1
+3 1 -5 0 0 3 1
+4 3 10 0 0 1 2
+5 3 -10 0 0 1 3
+6 3 0 5 0 1 1
+7 3 0 10 0 1 6
+''','''
+# Multipoint soma but not geometrically realizable
+#n,type,x,y,z,radius,parent
+1  1 0 0 0 10 -1
+2  1 2 0 0 10  1
+3  1 4 0 0 10  2
+4  3 2 1 0  1  2
+5  3 2 9 0  1  4
+''','''
+# 4pt soma with dendrite off pt 2
+#n,type,x,y,z,radius,parent
+1  1 0 0 0 10 -1
+2  1 3 0 0 10  1
+3  1 4 0 0 10  2
+4  1 6 0 0 10  3
+5  3 2 1 0  1  2
+6  3 2 9 0  1  5
+''','''
+# no soma 3 dendrites in a Y-shaped morphology, with lengths 10, 10, and 20
+#n,type,x,y,z,radius,parent
+1  3 0  0  0 1 -1
+2  3 10  0 0 1  1
+3  3 20  0 0 1  2
+4  3 10 20 0 1  2
+''','''
+# previous but with soma
+#n,type,x,y,z,radius,parent
+1  1 -11  0 0 5 -1
+2  1 -1  0  0 5  1
+3  3 0  0   0 1  2
+4  3 10  0  0 1  3
+5  3 20  0  0 1  4
+6  3 10 20  0 1  4
+''','''
+# 3 dendrites in a Y-shaped morphology, with lengths 10, 10, and 20
+#n,type,x,y,z,radius,parent (no soma)
+1  3 0  0  0 1 -1
+2  3 10  0 0 1  1
+3  3 11  0 0 1  2
+4  3 20  0 0 1  3
+5  3 10  1 0 1  2
+6  3 10 20 0 1  5
+''']
+
+def mkswc(swc_contents):
+  print ("\n\n\n")
+  f = open("temp.tmp", "w")
+  f.write(swc_contents)
+  f.close()
+  swc = h.Import3d_SWC_read()
+  swc.input("temp.tmp")
+  ig = h.Import3d_GUI(swc)
+  ig.instantiate(None)
+
+  print (swc_contents)
+  h.topology()
+  for sec in h.allsec():
+    r = [h.ref(0) for _ in range(3)]
+    style = ""
+    if sec.pt3dstyle():
+      h.pt3dstyle(1, r[0], r[1], r[2], sec=sec)
```

Comment:
sec.pt3dstyle(1, r[0], r[1], r[2])

Reply:
I don't think that signature has been implemented.

Reply:
It has. https://github.com/neuronsimulator/nrn/commit/4c65ee1ef242e709a30453f7bbe3a4e826add885

Reply:
I guess I'm not using it correctly then. I see that the 4 arg usage with h.ref(0) in the test file does not use the ```sec.pt3dstyle(....)``` form.
I don't see the return of a tuple from ```static PyObject* NPySecObj_pt3dstyle(NPySecObj* self, PyObject* args)```. Also the log message has the fragment
 ```
- sec.pt3dstyle does not support taking hoc refs because that's not Pythonic
...
- some Pythonic alternative should be provided to get the logical connection point.
  A reasonable option would be to define a new function and return a tuple.
```

Reply:
Oh. You're right. I misread this as setting not as reading the `pt3dstyle`.

I agree with past me that returning a tuple would be clearer. I opened issue #909 to discuss this further.